### PR TITLE
Switch to rolling upgrade

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/16697

### CODE REVIEW

**Review by corest — APPROVED** [review]
lgtm

File: helm/management-cluster-admission/templates/kustomize-out.yaml [hunk 1]
```diff
@@ -258,9 +258,9 @@ data:
     apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
     kind: ControllerManagerConfig
     health:
-      healthProbeBindAddress: :8081
+      healthProbeBindAddress: '{{ .Values.healthProbeBindAddress }}'
     metrics:
-      bindAddress: 127.0.0.1:8080
+      bindAddress: '127.0.0.1:{{ .Values.metricsBindAddress }}'
```
Comment (kopiczko) [inline helm/management-cluster-admission/templates/kustomize-out.yaml:263]: In general LGTM (even though @rossf7 said it won't help much) but I'm not sure why this is changed

**Comment by kubasobon** [comment]
@rossf7 @kopiczko I'm a bit confused :sweat_smile:
Q1) Will this change force `management-cluster-admission` to update using `Rolling` strategy? 
Q2) Will this change make something else worse?

**Comment by rossf7** [comment]
> Q1) Will this change force management-cluster-admission to update using Rolling strategy?

It won't change the deployment strategy for the deployment. But it's already using a rolling isn't it?

`helm upgrade --force` is disabled so the deployment will not be recreated.

> Q2) Will this change make something else worse?

The annotation name is confusing but it is hard to change as it's used in a lot of places.

With the value set to false chart-operator won't attempt an automated rollback. We do this for apps like coredns and nginx IC where the impact is high if they are down. 

https://github.com/giantswarm/chart-operator/blob/4732a9e1edc2ff117e885e6fa404d51d76f2f39c/service/controller/chart/resource/release/update.go#L226-L229

So with this change it's more likely there will be failed releases and you will get paged.

TLDR; Unless there is a problem with chart-operator using rollback I don't think this will help.

**Review by kopiczko — APPROVED** [review]

**Comment by kubasobon** [comment]
@rossf7 Thanks for breaking it down for me :)
Closing.
